tests/Test/Pager/Pagination/PaginatorTest.php

```diff
@@ -19,6 +19,19 @@ public function shouldThrowExceptionIfPageIsInvalid(): void
         $paginator->paginate(['a', 'b'], 0, -1);
     }
 
+    /**
+     * @test
+     */
+    public function shouldGetPageFromException(): void
+    {
+        try {
+            $paginator = $this->getPaginatorInstance();
+            $paginator->paginate(['a', 'b'], 0, -1);
+        } catch (PageNumberInvalidException $e) {
```

Comment:
Instead of catching the exception, we should assert that the exception is thrown

Reply:
Yes, that's why I added two tests for each exception.

One to assert the exception is thrown and another one to validate the return value of page/limit.